extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -112,15 +120,45 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
 
   /**
    * Triggers template update if opening without active product data.
-   * This happens when the "Shop Now" CTA is clicked.
+   * This happens either when the "Shop Now" CTA is clicked or when a
+   * shopping tag is clicked.
    * @param {boolean} isOpen
    * @private
    */
   onAttachmentStateUpdate_(isOpen) {
     if (!this.isOnActivePage_() || !isOpen) {
       return;
     }
+
     const shoppingData = this.storeService_.get(StateProperty.SHOPPING_DATA);
+
+    const productId =
+      Object.values(shoppingData[this.pageEl_.id])?.length === 1
+        ? Object.values(shoppingData[this.pageEl_.id])[0].productId
+        : shoppingData?.activeProductData?.productId;
+
+    if (productId === '') {
+      productId = 'products card listing page opened';
+    }
+
+    this.variableService_.onVariableUpdate(
+      AnalyticsVariable.STORY_SHOPPING_PRODUCT_ID,
+      productId
+    );
+    const shoppingItemClicked =
```

Comment:
`activeElement` doesn't work as expected, we would need sth else in it's place. Try this: long press on a tag and then print in the console `document.activeElement`, it would tell you the tag is the element. If you swipe up to open the PDP, the active element is still the tag because swiping doesn't change the `activeElement`. You could either update the variable service from the tag directly, or pass in the store state the element that was used to open the shopping attachment.

You're already updating this variable on the PLP click, so you just need that in the tag click

Reply:
Yes, I'll add the `SHOPPING_PRODUCT_ENGAGEMENT` event to the tag.